lib/index.js

```diff
@@ -29,17 +29,26 @@ function bind(func, obj) {
  * @param {String} name The name of the queue. Will be used to find abandoned queues and retry their items
  * @param {processFunc} fn The function to call in order to process an item added to the queue
  */
-function Queue(name, fn) {
+function Queue(name, opts, fn) {
+  if (typeof opts === 'function') fn = opts;
   this.name = name;
   this.id = uuid();
   this.fn = fn;
+  this.maxItems = opts.maxItems || Infinity;
+  this.maxAttempts = opts.maxAttempts || Infinity;
 
   this.timeouts = {
-    ACK_TIMER: 1000,
-    RECLAIM_TIMER: 3000,
-    RECLAIM_TIMEOUT: 10000,
-    RECLAIM_WAIT: 500,
-    MAX_QUEUE_DELAY: 30000
```

Comment:
Removing this is a breaking change I think. Should be ok to do so if we don't use it in a.js-integration-segmentio.